src/dataprotection/azext_dataprotection/manual/custom.py

```diff
@@ -139,6 +139,61 @@ def dataprotection_backup_instance_list_from_resourcegraph(client, datasource_ty
     return response.data
 
 
+def dataprotection_backup_instance_update_msi_permissions(cmd, client, resource_group_name, datasource_type, vault_name, operation, permissions_scope, backup_instance=None):
+    backup_vault = client.get(resource_group_name=resource_group_name,
+                vault_name=vault_name)
+    principal_id = backup_vault.identity.principal_id
+
+    role_assignments_arr = []
+
+    if operation == 'Backup' and backup_instance is None:
+        raise CLIError("Backup instance needs to be provided when operation is Backup")
```

Comment:
Make it "--backup-instance parameter is needed to be....."  so that the user understands what exactly is missing.